circuit-construction/src/writer.rs

```diff
@@ -789,6 +826,17 @@ impl<F: PrimeField> Cs<F> for System<F> {
             None
         }
     }
+
+    fn cached_constants(&mut self, x: F) -> Var<F> {
+        match self.cached_constants.get(&x) {
```

Comment:
What is the desired behaviour? Could you have different variables with coincidentally the same value?

Reply:
so when I want to create a variable that will hold a constant (like 3), I first check if I created such a variable in the past, and if I did I just return that instead. This way, the wiring will take care of ensuring that all of them are the same value.